pkg/controller/build/schedule_routine.go

```diff
@@ -158,13 +159,16 @@ func (action *scheduleRoutineAction) runBuild(ctx context.Context, build *v1.Bui
 		}
 
 		if lastTask || taskFailed {
+			log.Errorf(nil, "Unable to complete the build: %s", status.Error)
```

Comment:
Should it be logged when `lastTask` is truthy?

Reply:
Yes it is. I forgot to test the happy path *facelpalm*